### ISSUE TITLE
plugin: can not run a process that is in user's path

### ISSUE THREAD

**Issue by mulkieran** [cockpit-project/cockpit#3796]
The idea is:

1. Take the example ping plugin from plugin. Run it, just to be sure, all good.
2. Alter it so that line:
```
var proc = cockpit.spawn(["ping", "-c", "4", address.val()]);
```
is
```
var proc = cockpit.spawn(["test.py"]);
```
3. Ensure that there is a runnable python script, test.py, in user's bin.
```
[mulhern-journal@megadeth pinger]$ test.py
this is a test
[mulhern-journal@megadeth bin]$ which test.py
~/bin/test.py
[mulhern-journal@megadeth bin]$ echo $PATH
/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/mulhern-journal/.local/bin:/home/mulhern-journal/bin
```
4. Logout and log into cockpit as that user, i.e., mulhern-journal.
5. Expect to see "this is a test" instead of ping output, but instead get nothing and "Ooops" message.
6. Lots of output in javascript console, can't interpret at all.
7. Possibly relevant messages from journal:
```
Feb 22 11:22:24 megadeth.lab.bos.redhat.com sudo[38190]: mulhern-journal : user NOT in sudoers ; TTY=unknown ; PWD=/run/user/1002 ; USER=root ; COMMAND=/bin/cockpit-bridge --privileged
Feb 22 11:22:24 megadeth.lab.bos.redhat.com cockpit-bridge[38160]: sudo: a password is required
Feb 22 11:22:24 megadeth.lab.bos.redhat.com cockpit-bridge[38160]: This incident has been reported.
Feb 22 11:22:24 megadeth.lab.bos.redhat.com cockpit-bridge[38160]: Error executing command as another user: Not authorized
```
8. Decided to add myself to sudoers and not require a password,either.
```
[mulhern-journal@megadeth pinger]$ sudo ls
manifest.json  ping.html
```
Journal message is slightly changed as a result.
```
Feb 22 11:50:35 megadeth.lab.bos.redhat.com sudo[40305]: mulhern-journal : TTY=unknown ; PWD=/run/user/1002 ; USER=root ; COMMAND=/bin/cockpit-bridge --privileged
Feb 22 11:50:35 megadeth.lab.bos.redhat.com cockpit-bridge[40277]: This incident has been reported.
Feb 22 11:50:35 megadeth.lab.bos.redhat.com cockpit-bridge[40277]: Error executing command as another user: Not authorized
Feb 22 11:50:35 megadeth.lab.bos.redhat.com pkexec[40291]: mulhern-journal: Error executing command as another user: Not authorized [USER=root] [TTY=unknown] [CWD=/run/user/1002] [COMMAND=/usr/bin/cockpit-bridge --privileged]
Feb 22 11:50:35 megadeth.lab.bos.redhat.com polkitd[2344]: Operator of unix-session:c13 FAILED to authenticate to gain authorization for action org.freedesktop.policykit.exec for unix-process:40277:61565986 [cockpit-bridge] (owned by unix-user:mulhern-journal)
Feb 22 11:50:35 megadeth.lab.bos.redhat.com cockpit-bridge[40277]: cannot reauthorize identity(s): unix-user:root 
```

Tried putting test.py in /usr/local/bin, but similar failure in that case.

I feel that it is probably wrong for mulhern-journal running a script in mulhern-journal's own bin directory in mulhern-journal's own path to need to sudo in the first place, but I may be misunderstanding something.

P.S. My plugins are located in mulhern-journal's home directory ```/home/mulhern-journal/.local/share/cockpit```.

**Comment by stefwalter** [comment]
What about something like:

```
var proc = cockpit.spawn(["/bin/sh", "-c", "~/test.py"]);
```

**Comment by mulkieran** [comment]
It works! I don't have to be a sudoer or anything, either.

**Comment by mvollmer** [comment]
I don't whether it is related, but we recently fixed problems with $PATH in 8c1908035c919da58d29fab87b42126754f84405

**Comment by stefwalter** [comment]
@mulkieran Glad that works.

@mvollmer It would be nice way to pass a "directory" to the spawn stuff so that it can be the user's $HOME. Needs some more thought ... because it may be that other environment variables are interesting ...

Actually @petervo was thinking about ways to load environment variables into the ui, so that might help towards solving the $HOME stuff better.

I'll close this for now, but we should keep a better solution in mind.
